### PR TITLE
Added read/write function for whole result.

### LINKED ISSUE #588: Single load/save function for all pyPESTO result attributes
e.g.
```python
pypesto.store.write_result(pypesto_result: pypesto.Result, 'pypesto_result.hdf5')
pypesto_result = pypesto.store.read_result('pypesto_result.hdf5')
```
Should save all results if available i.e. optimization/profiling etc.

Can emit a warning for results that cannot be saved completely (e.g. `pypesto.Problem.objective`)

Possibly also optional toggles to not save certain attributes, e.g.
```python
write_result(pypesto_result, 'pypesto_result.hdf5', profile=False)
```

For the separate HDF5 files from optimization/profiling/etc, maybe generate them using pre-existing methods then combine into a e.g. ZIP archive?

### PR DESCRIPTION
**Pull request by PaulJonasJost**
Closes #588

### CODE REVIEW

File: test/base/test_store.py [hunk 1]
```diff
@@ -263,3 +263,97 @@ def test_storage_sampling():
...
             os.remove(fn)
+
+
+def test_storage_all():
+    """
+    This test tests the functions read_Result and write Result.
+    It currently does not test read/write of the problem as this
+    is know to not work completely. Also excludes testing history
+    key in optimization.
+    """
```
Comment (dilpath) [inline test/base/test_store.py:274]:
```suggestion
    """Test `read_result` and `write_result`.
    
    It currently does not test read/write of the problem as this
    is know to not work completely. Also excludes testing the history
    key of an optimization result.
    """
```

File: pypesto/store/read_from_hdf5.py [hunk 2]
```diff
@@ -256,3 +256,55 @@ def read(self) -> Result:
                                               parameter_id=parameter_id))
             self.results.profile_result.list = profiling_list
         return self.results
+
+
+def read_Result(filename: str,
```
Comment (yannikschaelte) [inline pypesto/store/read_from_hdf5.py:261]: I commented on the other function. all comments apply here too.

File: test/base/test_store.py [hunk 3]
```diff
@@ -263,3 +263,97 @@ def test_storage_sampling():
     finally:
         if os.path.exists(fn):
             os.remove(fn)
+
+
+def test_storage_all():
+    """
+    This test tests the functions read_Result and write Result.
```
Comment (yannikschaelte) [inline test/base/test_store.py:270]: typo

File: test/base/test_store.py [hunk 4]
```diff
@@ -263,3 +263,97 @@ def test_storage_sampling():
...
+    # Profiling
+    result = profile.parameter_profile(
+        problem=problem, result=result,
+        profile_index=[0], optimizer=optimizer)
+    # Sampling
+    x_0 = result.optimize_result.list[0]['x']
+    sampler = sample.AdaptiveParallelTemperingSampler(
+        internal_sampler=sample.AdaptiveMetropolisSampler(),
+        n_chains=1
+    )
```
Comment (dilpath) [inline test/base/test_store.py:298]: Could skip parallel tempering if only one chain?

File: test/base/test_store.py [hunk 5]
```diff
@@ -263,3 +263,97 @@ def test_storage_sampling():
     finally:
         if os.path.exists(fn):
             os.remove(fn)
+
+
+def test_storage_all():
+    """
+    This test tests the functions read_Result and write Result.
+    It currently does not test read/write of the problem as this
+    is know to not work completely. Also excludes testing history
```
Comment (yannikschaelte) [inline test/base/test_store.py:272]:
```suggestion
    is known to not work completely. Also excludes testing history
```

**Review by yannikschaelte — APPROVED** [review]
On a technical level looks all nice! Content-wise, I would delegate to @jvanhoefer @MerktSimon @dilpath who worked on the storage, whether all fits together as one would wish for.

**Review by dilpath — APPROVED** [review]
Thanks for this!

Some comments on the `read_Result` method would also apply to the `write_Result` method, if accepted.

**Pull request merged**